sdk/burdock/sql/private_rewriter.py

```diff
@@ -148,11 +150,16 @@ def exact_aggregates(self, query):
         select = Select(None, select)
 
         subquery = Query(child_scope.select(), query.source, query.where, query.agg, query.having, None)
-        subquery = self.per_key_random(subquery)
-        subquery = [AliasedRelation(subquery, "per_key_random")]
+        if self.options.reservoir_sample:
+            subquery = self.per_key_random(subquery)
+            subquery = [AliasedRelation(subquery, "per_key_random")]
 
-        filtered = Where(BooleanCompare(Column("per_key_random.row_num"), "<=", Literal(str(self.max_contrib), self.max_contrib)))
-        return Query(select, From(subquery), filtered, query.agg, None, None)
+            filtered = Where(BooleanCompare(Column("per_key_random.row_num"), "<=", Literal(str(self.options.max_contrib), self.options.max_contrib)))
+            return Query(select, From(subquery), filtered, query.agg, None, None)
+        else:
+            subquery = self.per_key_clamped(subquery)
+            subquery = [AliasedRelation(subquery, "per_key_all")]
+            return Query(select, From(subquery), None, query.agg, None, None)
```

Comment:
For databricks spark, how to return 

"SELECT .... FROM (subquery_table) ..."

instead of a subquery string, just a string representing the return data set, as in databricksSparkReader, I cache the subquery result in a temp view table.

I think you're working on this?

Reply:
Yep, that will be in a future PR